src/extensions/askUserForParam.ts

```diff
@@ -116,6 +129,81 @@ export async function askForParam(paramSpec: Param): Promise<string> {
   return response;
 }
 
+async function promptSecret(
+  projectId: string,
+  instanceId: string,
+  paramSpec: Param
+): Promise<string> {
+  if (paramSpec.reconfiguring) {
+    const action = await promptOnce({
+      type: "list",
+      message: `Choose what you would like to do with this secret:`,
+      choices: [
+        { name: "Leave unchanged", value: SecretUpdateAction.LEAVE },
+        { name: "Set new value", value: SecretUpdateAction.SET_NEW },
+      ],
+    });
+    switch (action) {
+      case SecretUpdateAction.SET_NEW:
```

Comment:
Seems like a simple if statement would be a lot cleaner than this case statement here - is this so we can support more varied types of secret management in the future?

Reply:
Yeah, I think we might add a third option at some point. Something like "unset" or "reset" -- in cases when param is optional and they want to make it blank. For now this case will work by allowing them to set empty value in the secret payload.